optuna/distributions.py

```diff
@@ -297,6 +295,72 @@ def _contains(self, param_value_in_internal_repr):
         return self.low <= value <= self.high
 
 
+class IntLogUniformDistribution(BaseDistribution):
```

Comment:
I personally prefer extending the existing `IntDistribution` class instead of creating a new class (c.f. the `step` field added by https://github.com/optuna/optuna/pull/910).
What do you think of it?

Reply:
Ah, I'm really sorry that I suggested implementing `IntLogUniformDistribution` [here](https://github.com/optuna/optuna/issues/510#issuecomment-622399411) without understanding [the historical context](https://github.com/optuna/optuna/issues/510#issuecomment-622867398). I support @sile since I'm not confident about the internal API design.

Really sorry @nzw0301 for confusing you.

Reply:
I see. Thank you for your explanation.

Reply:
@sile  Thank you for your comment! It sounds great. I will update it.

@himkt Never mind :)

Reply:
Sorry for the late comment, but please let me confirm that did you decide to keep `IntLogUniformDistribution` class? (instead of adding `log` field to the existing `IntDistribution` as I suggested).

Reply:
@sile I apologise that I completely misunderstood your comment. I will fix it.

Reply:
@nzw0301 Thank you for your reply. Honestly, now I'm not confident that it's definitely better to extend the existing class...
I'd like to hear opinions from other dev members. What do you think of it? > @HideakiImamura @toshihikoyanase

Reply:
To be honest, me neither. Note that If we extend `IntUniformDistribution`, the same problem pointed at  https://github.com/optuna/optuna/pull/1201#discussion_r419170507 happens. As a result, I will make `log` and `step` keyword-only arguments or add `log ` argument after `step` argument but it is not consistent with `suggest_int` function.

Reply:
Again, sorry for joining in the middle of the conversation but I think there's room for rethinking what concrete `BaseDistributions` we should provide and what parameters they should involve, now that the suggest APIs are being changed. I'm wondering if we shouldn't do that first before discussing this particular class and leave this PR as is. C.f. https://github.com/optuna/optuna/issues/510

Reply:
@nzw0301 @hvy Thank you for your feedback! Now, I think that it's better to keep this PR as is, so let me resolve this conversation.